wlroots/ffi_build.py

```diff
@@ -354,10 +359,9 @@
         struct wl_signal precommit;
         struct wl_signal commit;
         struct wl_signal present;
+        struct wl_signal bind;
         struct wl_signal enable;
         struct wl_signal mode;
-        struct wl_signal scale;
-        struct wl_signal transform;
```

Comment:
I noticed this in the wlroots changelog. I have been looking for a way to have cffi check for correctness on these nested structs using something like `...;` as we can do for the top level structs, but the c parser doesn't like these being partial. If you can find a way for this to be caught at compile time, I'd really appreciate it.

Reply:
Not sure I follow -- what needs to be checked? That the size of that struct matches how wlroots defines it?

Reply:
It is not just the size, but also the name and types of the struct members. For example if you change the name of `events`, then you will get a compilation failure, but here the names and count of member variables is different and there seemingly is no error or warning messages generated. I guess really just something that would have flagged this as a problem at the point where it is building the cffi module would be a great step.

Reply:
I can't seem to find anything that could trigger an error here. Interestingly it won't even error if two of the structs have the same name. Perhaps it could be worth looking into automatically generating the CDEF string from the wlroots headers?